tenseal/tensors/ckksvector.cpp

```diff
@@ -176,13 +198,23 @@ CKKSVector& CKKSVector::mul_inplace(CKKSVector to_mul) {
         throw invalid_argument("can't multiply vectors of different sizes");
     }
 
+    if (this->context->is_auto_mod_switch() &&
+        this->ciphertext.parms_id() != to_mul.ciphertext.parms_id()) {
+        set_to_same_mod(this->context, this->ciphertext, to_mul.ciphertext);
+    }
+
     this->context->evaluator->multiply_inplace(this->ciphertext,
                                                to_mul.ciphertext);
 
-    // TODO: can make this optional
-    // relineraize after ciphertext-ciphertext multiplication
-    this->context->evaluator->relinearize_inplace(this->ciphertext,
-                                                  this->context->relin_keys());
+    if (this->context->is_auto_relin()) {
+        this->context->evaluator->relinearize_inplace(
+            this->ciphertext, this->context->relin_keys());
+    }
+
+    if (this->context->is_auto_rescale()) {
+        this->context->evaluator->rescale_to_next_inplace(this->ciphertext);
+        this->ciphertext.scale() = this->init_scale;
```

Comment:
Should we make init_scale `const double`?

Keep in mind, this won't work for consts in constructors
`A::A(double scale){
   this->init_scale = scale;
}`

Instead, you have to leverage the initializer lists mechanism.
``A::A(double scale) : init_scale(scale){}``

Reply:
but the constructor doesn't just put the scale value there, there is some checks before.